expo/eas-cli
# Add EXPO_APPLE_TEAM_ID and EXPO_APPLE_PROVIDER_ID support

# Why

Added support for `EXPO_APPLE_PROVIDER_ID` and `EXPO_APPLE_TEAM_ID` when authenticating without API Key enabled. This means users should be able to skip all prompts (assuming 2FA is still valid). `EXPO_APPLE_PROVIDER_ID` is cookies-auth only.

## Teams vs Providers

- The Apple Provisioning API (Bundle IDs, Capabilities, Certificates, Provisioning Profiles) requires a **Team ID** -- Accessed via the `Teams` module.
- The Iris API (Apps, Screenshots, Users, Invites) requires a **Provider ID** -- Accessed via the `Session` module.

### Teams

Teams are associated with the Portal API and indicate whether a user has the ability to provision new data. The team is often indicated to the Apple server

### Providers

Providers are associated with the user via cookies that get mutated when `/olympus/v1/session` is posted to with a provider id.

# Test Plan

- Run `eas credentials -p ios` -> note the provider id and team id.
- Set the environment variables.
- Run `eas credentials -p ios` and note that all the prompts are skipped.
